engine/schema/src/main/java/com/cloud/upgrade/dao/Upgrade41520to41600.java

```diff
@@ -65,6 +68,31 @@ public boolean supportsRollingUpgrade() {
     @Override
     public void performDataMigration(Connection conn) {
         generateUuidForExistingSshKeyPairs(conn);
+        fixWrongPoolUuid(conn);
+    }
+
+    public void fixWrongPoolUuid(Connection conn) {
+        LOG.debug("Replacement of faulty pool uuids");
+        try (PreparedStatement pstmt = conn.prepareStatement("SELECT id,uuid FROM storage_pool "
+                + "WHERE removed IS NULL;"); ResultSet rs = pstmt.executeQuery()) {
+            PreparedStatement updateStmt = conn.prepareStatement("update storage_pool set uuid = ? where id = ?");
+            while (rs.next()) {
+                if (!rs.getString(2).contains("-")) {
```

Comment:
can we make this part of the where-clause of the select statement already? no reason to try and fix anything that isn't broken. (also is this condition good and enough?)

Reply:
Hi @DaanHoogland, hmm not quite sure what do you mean. I'm updating only uuids with the wrong format or do you want to reduce the result set of this query with a regular expression 
`SELECT id,uuid FROM storage_pool  WHERE removed IS NULL;`
If yes, I'm not really a fan of regular expressions in sql queries and I try to prevent this if it not absolutely necessary. 
The result set should be not so big, therefore this solution. But I can change this if it requested.

I'm check out the storage pool table and in my opinion the conditions are right.

Reply:
not vital, /me nitpicking. I meant to move the contains check on the string to the sql to reduce the result set . i.e. `SELECT id,uuid FROM storage_pool "
                + "WHERE removed IS NULL and uuid not LIKE '%-%-%' ` or some like it. please resolve if you don't feel like it, this should only be run once (on eache installation anyway.